### DISCUSSION TITLE
GridGeoSampler on areas smaller than patch size

### DISCUSSION THREAD

**Discussion by mpelchat04** [torchgeo/torchgeo#1624] [Q&A]
We use the GridGeoSampler to create patches. Although, we have cases where the portion covered by the training data is small (smaller than patch_size), as seen in pink in the image:  
![bcf04b15-c161-45de-b9f8-d2e60ca1f8c4](https://github.com/microsoft/torchgeo/assets/38693210/6b5e86c8-cbc7-459e-afe8-a7c172e8d495)
The image and data showed here are taken from the Spacenet-7 dataset.  
Using the GridGeoSampler, these patches are discarded and it seems like an intended feature [ref](https://github.com/microsoft/torchgeo/issues/431#issuecomment-1049503005).  
Is there a logical way to overcome this limitation, where it discards useful patches?

**Comment by adamjstewart** [comment]
I think this should be easy to fix. The only thing is deciding how we want it to behave.

Do we want all samplers to return all intersections? What if the area of intersection is 0? What if we're using a point dataset?

Can you open an issue so we can track this for the next release? Otherwise we can keep discussing what this should look like in this discussion.
